### PR TITLE
Add the 'hold' feature as requested by #18

### PR DESCRIPTION
**Pull request by ctxnop**
Select the 'Hold' value in the 'Click type'. It disable widgets that
make no sense in such situation. Emits the 'mouse down' event on start,
and 'mouse up' on stop.

Signed-off-by: ctxnop <ctxnop@gmail.com>

### CODE REVIEW

File: src/x11api.c [hunk 1]
```diff
@@ -88,49 +88,73 @@ int cxevent(Display *display, long mask, XButtonEvent event)
...
+        XQueryPointer(display, event.window,
+                        &event.root, &event.subwindow,
+                        &event.x_root, &event.y_root,
+                        &event.x, &event.y,
+                        &event.state);
+    }
+    return event;
+}
+
+int mouse_down(Display *display, int button, int mode)
```
Comment (robiot) [inline src/x11api.c:111]: Why have 2 almost identical functions. Instead make one called something like mouse_event, then it accepts an enum value of UP or DOWN.

```c
enum MouseEvents
{
    MOUSE_UP,
    MOUSE_DOWN,
};


int mouse_event(..., enum MouseEvents event_type) {
}
```

**Comment by robiot** [comment]
I can fix these since I'm kinda bored atm

File: src/x11api.c (hunk 1 above)
Comment (ctxnop) [reply src/x11api.c:111]: you're right. I should have think about that. If you don't fix it, I'll tomorrow. Or this night if I can't sleep for whatever reason.

**Comment by ctxnop** [comment]
I fixed it. Let me know if you see any other issue I can solve.

**Comment by robiot** [comment]
LGTM. Will check again tomorrow and hopefully merge.
It would be good if there is any good way of running the press down event once and then skip the loop so it doesn't have to try to run it multiple times per second. But I can't really think of a way that is clean. So lets keep it like this for now :)

**Comment by ctxnop** [comment]
Yeah, I agree. I thought that the main thread could lock a mutex on start and release it on stop. So the "clicking" thread would lock the same mutex and be blocked until the main thread released it. But, to be honest, I added this feature because I needed it this afternoon for few hours only. So I didn't want to spend too much time on it. If you want me to implement something using locks, tell me, I'll propose you something soon.

File: src/x11api.h [hunk 2]
```diff
@@ -7,17 +7,19 @@
...
 /**
  * @brief Enumeration of the supported mouse events.
  * @see mouse_event
  */
 enum MouseEvents
 {
-    MOUSE_PRESSED,
-    MOUSE_RELEASED
+    MOUSE_EVENT_PRESS,
+    MOUSE_RELEASE,
```
Comment (ctxnop) [inline src/x11api.h:23]: Feels a bit weird to me. Why there is a `_EVENT_` part for press but not for release?

Comment (robiot) [reply src/x11api.h]: Thank you for pointing that out. Didn't notice it.

**Review by ctxnop — COMMENTED** [review]
LGTM. Maybe adding some braces on if/else branches. I'm used to omit those but that may not fit your coding styles.

**Pull request merged**